official/recommendation/ncf_keras_main.py

```diff
@@ -189,11 +191,25 @@ def _get_keras_model(params):
   return keras_model
 
 
+def _get_keras_optimizer(params):
+  """Returns the optimizer used by the keras model."""
+  optimizer = tf.keras.optimizers.Adam(
+      learning_rate=params["learning_rate"],
+      beta_1=params["beta1"],
+      beta_2=params["beta2"],
+      epsilon=params["epsilon"])
+  if params["use_tpu"]:
```

Comment:
I dont think this else clause is useful in the keras codepath yet. when we add TPUStrategy to this codepath, even then it would not need any such wrapping. You probably need to move this else clause to the estimator path's model_fn which is the route that TPUEstimator takes